src/scripts/main.js

```diff
@@ -1,3 +1,53 @@
 'use strict';
 
-// write code here
+// Функция для сортировки сотрудников по зарплате
+const sortEmployeesBySalary = function (selector) {
+  // Получаем все элементы списка по переданному селектору
+  const employeeItems = document.querySelectorAll(selector);
+
+  // Преобразуем NodeList в массив и сортируем его по зарплате
+  const sortedEmployee = Array.from(employeeItems).sort((a, b) => {
+    // Извлекаем зарплату из атрибутов данных и преобразуем в число
+    const salaryA = parseInt(a.dataset.salary.replace(/[$,]/g, ''), 10);
+    const salaryB = parseInt(b.dataset.salary.replace(/[$,]/g, ''), 10);
+
+    return salaryB - salaryA; // Сортируем по убыванию
+  });
+
+  return sortedEmployee; // Возвращаем отсортированный массив сотрудников
```

Comment:
This function `sortEmployeesBySalary` modifies the DOM directly by sorting DOM elements. It's recommended to separate the logic of data processing and UI updates. Consider refactoring this function to only sort an array of employee objects by salary, and create a separate function to update the DOM based on this sorted array.